File: src/main/java/com/flexpoker/table/command/factory/TableFactory.java
package com.flexpoker.table.command.factory;

import java.util.List;

import com.flexpoker.table.command.aggregate.Table;
import com.flexpoker.table.command.commands.CreateTableCommand;
import com.flexpoker.table.command.framework.TableEvent;

public interface TableFactory {

    Table createNew(CreateTableCommand command);

    Table createFrom(List<TableEvent> events);

}

File: src/main/kotlin/com/flexpoker/table/command/factory/TableFactory.kt
package com.flexpoker.table.command.factory

import com.flexpoker.table.command.aggregate.Table
import com.flexpoker.table.command.commands.CreateTableCommand
import com.flexpoker.table.command.events.TableEvent

interface TableFactory {
    fun createNew(command: CreateTableCommand): Table
    fun createFrom(events: List<TableEvent>): Table
}